frontend/src/routes/create-goals/+page.svelte

```diff
@@ -49,170 +49,275 @@
       console.log(error);
     }
   }
+
+  let showModal = false; // Modal state
+
+  const openModal = () => {
+    showModal = true;
+  };
+
+  const closeModal = () => {
+    showModal = false;
+  };
+
+  const handleEdit = () => {
+    // Handle the edit logic here...
+    closeModal();
+  };
 </script>
 
-<div class="flex">
+<div class="page-fade-in">
   <div class="divider" />
-</div>
-<div class="container mx-auto px-4 pt-5 md:max-w-xl lg:max-w-3xl z-10">
-  <div class="mb-5">
-    <label class="block text-xl font-bold mb-2" for="goal">Goal</label>
-    <input
-      class="shadow appearance-none border rounded w-full py-2 px-3 leading-tight focus:outline-none focus:shadow-outline"
-      id="goal"
-      type="text"
-      placeholder="What's your SMART Goal?"
-      bind:value={goal.goal}
-    />
-    <ErrorMessage
-      errorMessageId="goal-error"
-      errorMessage="SMART goal is required"
-      showError={goalError}
-    />
-
-    <div class="mt-3 flex flex-col items-left">
-      <button id="generate" class="btn btn-primary">Generate</button>
-      {#if loadingGenerate}
-        <div class="mt-3 flex justify-center items-center">
-          <span class="loading loading-infinity loading-md" />
-        </div>
-      {/if}
-    </div>
-  </div>
 
-  <div class="grid grid-cols-1 md:grid-cols-2 gap-5">
-    <!-- Specific card -->
-    <div class="card bordered">
-      <figure>
-        <figcaption class="p-4 card-body">
-          <h2 class="card-title">Specific</h2>
-          <input
-            id="specific"
-            class="shadow appearance-none border rounded w-full py-2 px-3 leading-tight focus:outline-none focus:shadow-outline"
-            type="text"
-            placeholder="AchievAIm's Specific suggestion"
-            bind:value={goal.specific}
-          />
-          <div class="mt-3">
-            <button class="btn btn-primary">Keep Specific Suggestion</button>
-          </div>
-        </figcaption>
-      </figure>
-    </div>
+  <div class="container shadow-lg rounded-xl mb-4 mx-auto px-4 pt-5 md:max-w-xl lg:max-w-3xl z-10">
```

Comment:
All the fields on this form can be replaced with the component.

Reply:
Okay. I created frontend\src\lib\components\EditGoalComponent.svelte that replaces this. The Modal element is the only time left out since this needs to be different for different pages. See #256